### ISSUE TITLE
Inefficiencies causing slow load times for various screens  

### ISSUE THREAD

**Issue by zeno-mcdohl** [ccrama/Slide#3262]
Slide version: 6.5
Android version: 7.0

From https://www.reddit.com/r/slideforreddit/comments/huncs7/is_there_a_known_issue_with_loading_speed/fyoa9rl/

There are various pages that hang while loading and take 10-30 seconds to load. A contributor to the project describes this as being caused by inefficient method around checking which posts have been read, and poor handling of background tasks like usernote checking.

I did not see an issue submitted for this, so I have submitted this one.

**Comment by TacoTheDank** [comment]
@zeno-mcdohl Yeah, this app is in dire need of an efficiency overhaul. Good news is that ccrama is actually working on Slide v2, which will be completely written from scratch in Kotlin. So we'll see if that same issue arises there.
